### PR TITLE
3208 - Fix a locale issue loading en-GB

### LINKED ISSUE #3208: Locale: Getting wrong data in the locale
**Describe the bug**
We are seeing the wrong locale being used in some cases. For example en-GB is using en0US settings. Possible timing issue

**To Reproduce**
1. Go to https://jsfiddle.net/ekonfdc1/
2. Should be using dd/MM/yyyy for en-GB
3. But it is not (note it may work initially but if you reload the page you will see the wrong locale)
4. Also not that en-US and en-GB scripts are appended more than one time.

**Expected behavior**
Should use en-GB if specified.

**Version**
- ids-enterprise: [e.g. v4.24.0 ]

**Platform**
 - Chrome

**Additional context**
Guessing either #2368 or #2238 were the ones which caused the issue.

### PR DESCRIPTION
**Pull request by tmcconechy**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

When loading en-Gb some of the data from en-US was getting mixed in. Also multiple of the same scripts are loaded into the page.

**Related github/jira issue (required)**:
Fixes #3208 

**Steps necessary to review your pull request (required)**:
- go to http://localhost:4000/components/locale/test-format-date
- should show the respected locales's formats for dates (04/12/2019, 12/4/2019, 04-12-2019)
- also covered by (passing) tests
- spot check sample app, including setting locales.

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: test/components/locale/locale.func-spec.js [hunk 1]
```diff
@@ -90,6 +90,30 @@ describe('Locale API', () => {
...
+
+    locale = 'da-DK';
+    Locale.set(locale).done((name) => {
+      expect(name).toEqual('da-DK');
+      expect(Locale.currentLocale.data.englishName).toEqual('Danish (Denmark)');
+      expect(Locale.formatDate(new Date(2019, 11, 4))).toEqual('04-12-2019');
+      done();
+    });
+  });
+
```
[the remark this answers is not in the document]
Comment (nbcp) [reply test/components/locale/locale.func-spec.js]: I would think an e2e test on the new test-format-date.html would be a better alternative for this

**Comment by tmcconechy** [comment]
Ok, i add an e2e test as well (which did fail on the old code). Kept the new functional test in. I think it has some use.

**Comment by fitzorama** [comment]
@tmcconechy We will need this patched back to 4.23.0-rc.0 please.

**Comment by tmcconechy** [comment]
We cannot patch an rc release. I can make a 4.23.1 however which will be all the changes up from rc.0 to 4.23 final.

**Comment by fitzorama** [comment]
Yes, I knew I stated that incorrectly. Is there a way to know what the changes were between 4.23.0-rc.0 and 4.23.1?

**Comment by tmcconechy** [comment]
Its going to be all the changes in this list https://github.com/infor-design/enterprise/commits/4.23.x from "Commits on Nov 14, 2019" up

**Comment by fitzorama** [comment]
Yes, I have tested it and am seeing some issues. Soho.Locale.currentLanguage is undefined. 

Landmark was checking the Locale.currentLanguage.name to set the extended messages. And the datepicker is throwing errors as it also is for checking Locale.currentLanguage.name.

image

**Comment by tmcconechy** [comment]
How can i review that error? Is it just loading this in the demo app for NG? Something in datepicker?

**Comment by fitzorama** [comment]
If you bring sohoxi.js into ng, the locale-initializer.service will fail as it is checking for Locale.currentLanguage.name as well.

**Comment by tmcconechy** [comment]
@fitzorama - are you still seeing any issues, im unable to reproduce that on my end. The test works immediately

**Comment by fitzorama** [comment]
I asked @nbcp to test with Landmark and his testing was successful, so I'm good.

**Comment by fitzorama** [comment]
@tmcconechy Your latest push is working well.

**Comment by nbcp** [comment]
Hi @tmcconechy, as I was making some e2e tests for ep-ng, I found a similar issue still occuring for **pt-BR** even with your latest changes. This can be replicated easily using http://localhost:4200/ids-enterprise-ng-demo/locale-pipe

During first load, it's setting the data from **pt-PT** instead of **pt-BR**. Both **pt-PT** and **pt-BR** culture files are appended. See below screenshot: 

- left shows expected correct date, number, and currency format. 
- right shows the issue during selection of pt-BR locale after fresh reload 
image
image


So I'm thinking probably this issue also happens for other locales like: `ar-*`, `es-*`, `fr-*`, `zh-*`

**Comment by tmcconechy** [comment]
Thanks @nbcp  i see this and added an additional fix.

File: src/components/locale/locale.js [hunk 2]
```diff
@@ -294,6 +302,11 @@ const Locale = {  // eslint-disable-line
         this.setCurrentLocale(parentLocale, this.cultures[parentLocale]);
         this.dff[parentLocale].resolve(parentLocale);
       }
+      if (parentLocale && this.dff[locale] && this.cultures[locale]) {
+        console.log(parentLocale, locale);
```
Comment (nbcp) [inline src/components/locale/locale.js]: I think this should've been removed

**Pull request merged**